common/common_test.go

```diff
@@ -670,11 +670,6 @@ func TestNilGuard(t *testing.T) {
 
 	err = NilGuard(&s, nil, (*int)(nil))
 	assert.ErrorIs(t, err, ErrNilPointer)
-	assert.ErrorContains(t, err, "*int")
-	var mErr *multiError
-	require.ErrorAs(t, err, &mErr, "err must be a multiError")
-	assert.Len(t, mErr.Unwrap(), 2, "Should get 2 errors back")
-
```

Comment:
🚧 Change request:
* I agree that you needed to change this,  I think you need to keep the assert on type, and switch around the args so you get that. In other words: We want to cover that we get past a non-nil input and find a subsequent error
* I think you need coverage of not blowing up
```suggestion
    type nilWrap struct {
        i *int
    }
    var nilWrap *nilWrap
	err = NilGuard(&s, &nilWrap, nilWrap.i)
	assert.ErrorIs(t, err, ErrNilPointer)
	assert.ErrorContains(t, err, "*nilWrap")
```

Note, Haven't tested this at all. It's just communicating my feedback in code.